src/App.tsx

```diff
@@ -1,61 +1,120 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './components/types';
+import { TodoList } from './components/TodoList';
+
+function getUserById(id: number) {
+  return usersFromServer.find(user => user.id === id) || null;
+}
+
+function getHighestTodoId(list: Todo[]) {
+  const listOfId = list.map(item => item.id);
+
+  return Math.max(...listOfId);
+}
+
+const defaultList = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(defaultList);
+
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setTitleError(!title.trim());
+    setUserError(!userId);
+
+    if (!title.trim() || !userId) {
+      // eslint-disable-next-line
+      return;
+    }
+
+    setTodos(currentList => [
+      ...currentList,
+      {
+        id: getHighestTodoId(currentList) + 1,
```

Comment:
Read about [crypto.randomUUID()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) to use it in future when you need a random id